<a href="https://colab.research.google.com/github/Aleksandera0o/ML_VVSU/blob/main/lab1/%D0%97%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D0%B5_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Задание по практике 1

##Загрузка датасета
В файле classification datasets расположены ссылки на датасеты. Выберите один в соответствии со своим вариантом, скачайте и прочитайте файл

**Не забудьте загрузить нужные библиотеки**

Например: pip3 install pandas

In [ ]:
import pandas as pd

Нет ответа не следующие вопросы: "Какие средние значения в каждой числовой колонке вашего датасета?
Минимальные и максимальные значения в каждой числовой колонке вашего датасета
"
Типизировать надо все колонки
Нет ответа на вопрос: "Какие признаки вы можете выделить в своем датасете (категориальные, числовые и др., минимум 3 разных, помните что можно также выделить признаки более точно, например числовые - порядковые, количественные и т.д.)?"

In [ ]:
df = pd.read_csv('/content/Global_YouTube_Statistics.csv', encoding = "ISO-8859-1")
df

rank                    Youtuber  subscribers   video views  \
0       1                    T-Series    245000000  2.280000e+11   
1       2              YouTube Movies    170000000  0.000000e+00   
2       3                     MrBeast    166000000  2.836884e+10   
3       4  Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
4       5                   SET India    159000000  1.480000e+11   
..    ...                         ...          ...           ...   
990   991               Natan por Aï¿     12300000  9.029610e+09   
991   992    Free Fire India Official     12300000  1.674410e+09   
992   993                       Panda     12300000  2.214684e+09   
993   994                 RobTopGames     12300000  3.741235e+08   
994   995                Make Joke Of     12300000  2.129774e+09   

             category                       Title  uploads         Country  \
0               Music                    T-Series    20082           India   
1    Film & Animation               youtubemovies        1   United States   
2       Entertainment                     MrBeast      741   United States   
3           Education  Cocomelon - Nursery Rhymes      966   United States   
4               Shows                   SET India   116536           India   
..                ...                         ...      ...             ...   
990            Sports               Natan por Aï¿     1200          Brazil   
991    People & Blogs    Free Fire India Official     1500           India   
992               NaN                 HybridPanda     2452  United Kingdom   
993            Gaming                 RobTopGames       39          Sweden   
994            Comedy                Make Joke Of       62           India   

    Abbreviation   channel_type  ...  subscribers_for_last_30_days  \
0             IN          Music  ...                     2000000.0   
1             US          Games  ...                           NaN   
2             US  Entertainment  ...                     8000000.0   
3             US      Education  ...                     1000000.0   
4             IN  Entertainment  ...                     1000000.0   
..           ...            ...  ...                           ...   
990           BR  Entertainment  ...                      700000.0   
991           IN          Games  ...                      300000.0   
992           GB          Games  ...                        1000.0   
993           SE          Games  ...                      100000.0   
994           IN         Comedy  ...                      100000.0   

     created_year  created_month  created_date  \
0          2006.0            Mar          13.0   
1          2006.0            Mar           5.0   
2          2012.0            Feb          20.0   
3          2006.0            Sep           1.0   
4          2006.0            Sep          20.0   
..            ...            ...           ...   
990        2017.0            Feb          12.0   
991        2018.0            Sep          14.0   
992        2006.0            Sep          11.0   
993        2012.0            May           9.0   
994        2017.0            Aug           1.0   

     Gross tertiary education enrollment (%)    Population  Unemployment rate  \
0                                       28.1  1.366418e+09               5.36   
1                                       88.2  3.282395e+08              14.70   
2                                       88.2  3.282395e+08              14.70   
3                                       88.2  3.282395e+08              14.70   
4                                       28.1  1.366418e+09               5.36   
..                                       ...           ...                ...   
990                                     51.3  2.125594e+08              12.08   
991                                     28.1  1.366418e+09               5.36   
992                                     60.0  6.683440e+07               3.85   

##Первое знакомство с датасетом
В следующих ячейках выведите:


1.   Колонки датасета
2.   Первые 5 значений
3.   Последние 5 значений
4.   Статистику по числовым колонкам



In [ ]:
df.columns

Index(['rank', 'Youtuber', 'subscribers', 'video views', 'category', 'Title',
       'uploads', 'Country', 'Abbreviation', 'channel_type',
       'video_views_rank', 'country_rank', 'channel_type_rank',
       'video_views_for_the_last_30_days', 'lowest_monthly_earnings',
       'highest_monthly_earnings', 'lowest_yearly_earnings',
       'highest_yearly_earnings', 'subscribers_for_last_30_days',
       'created_year', 'created_month', 'created_date',
       'Gross tertiary education enrollment (%)', 'Population',
       'Unemployment rate', 'Urban_population', 'Latitude', 'Longitude'],
      dtype='object')

In [ ]:
df.head(5)

rank                    Youtuber  subscribers   video views  \
0     1                    T-Series    245000000  2.280000e+11   
1     2              YouTube Movies    170000000  0.000000e+00   
2     3                     MrBeast    166000000  2.836884e+10   
3     4  Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
4     5                   SET India    159000000  1.480000e+11   

           category                       Title  uploads        Country  \
0             Music                    T-Series    20082          India   
1  Film & Animation               youtubemovies        1  United States   
2     Entertainment                     MrBeast      741  United States   
3         Education  Cocomelon - Nursery Rhymes      966  United States   
4             Shows                   SET India   116536          India   

  Abbreviation   channel_type  ...  subscribers_for_last_30_days  \
0           IN          Music  ...                     2000000.0   
1           US          Games  ...                           NaN   
2           US  Entertainment  ...                     8000000.0   
3           US      Education  ...                     1000000.0   
4           IN  Entertainment  ...                     1000000.0   

   created_year  created_month  created_date  \
0        2006.0            Mar          13.0   
1        2006.0            Mar           5.0   
2        2012.0            Feb          20.0   
3        2006.0            Sep           1.0   
4        2006.0            Sep          20.0   

   Gross tertiary education enrollment (%)    Population  Unemployment rate  \
0                                     28.1  1.366418e+09               5.36   
1                                     88.2  3.282395e+08              14.70   
2                                     88.2  3.282395e+08              14.70   
3                                     88.2  3.282395e+08              14.70   
4                                     28.1  1.366418e+09               5.36   

   Urban_population   Latitude  Longitude  
0       471031528.0  20.593684  78.962880  
1       270663028.0  37.090240 -95.712891  
2       270663028.0  37.090240 -95.712891  
3       270663028.0  37.090240 -95.712891  
4       471031528.0  20.593684  78.962880  

[5 rows x 28 columns]

In [ ]:
df.tail(5)

rank                  Youtuber  subscribers   video views  \
990   991             Natan por Aï¿     12300000  9.029610e+09   
991   992  Free Fire India Official     12300000  1.674410e+09   
992   993                     Panda     12300000  2.214684e+09   
993   994               RobTopGames     12300000  3.741235e+08   
994   995              Make Joke Of     12300000  2.129774e+09   

           category                     Title  uploads         Country  \
990          Sports             Natan por Aï¿     1200          Brazil   
991  People & Blogs  Free Fire India Official     1500           India   
992             NaN               HybridPanda     2452  United Kingdom   
993          Gaming               RobTopGames       39          Sweden   
994          Comedy              Make Joke Of       62           India   

    Abbreviation   channel_type  ...  subscribers_for_last_30_days  \
990           BR  Entertainment  ...                      700000.0   
991           IN          Games  ...                      300000.0   
992           GB          Games  ...                        1000.0   
993           SE          Games  ...                      100000.0   
994           IN         Comedy  ...                      100000.0   

     created_year  created_month  created_date  \
990        2017.0            Feb          12.0   
991        2018.0            Sep          14.0   
992        2006.0            Sep          11.0   
993        2012.0            May           9.0   
994        2017.0            Aug           1.0   

     Gross tertiary education enrollment (%)    Population  Unemployment rate  \
990                                     51.3  2.125594e+08              12.08   
991                                     28.1  1.366418e+09               5.36   
992                                     60.0  6.683440e+07               3.85   
993                                     67.0  1.028545e+07               6.48   
994                                     28.1  1.366418e+09               5.36   

     Urban_population   Latitude  Longitude  
990       183241641.0 -14.235004 -51.925280  
991       471031528.0  20.593684  78.962880  
992        55908316.0  55.378051  -3.435973  
993         9021165.0  60.128161  18.643501  
994       471031528.0  20.593684  78.962880  

[5 rows x 28 columns]

In [ ]:
df.describe()

rank   subscribers   video views        uploads  video_views_rank  \
count  995.00000  9.950000e+02  9.950000e+02     995.000000      9.940000e+02   
mean   498.00000  2.298241e+07  1.103954e+10    9187.125628      5.542489e+05   
std    287.37606  1.752611e+07  1.411084e+10   34151.352254      1.362782e+06   
min      1.00000  1.230000e+07  0.000000e+00       0.000000      1.000000e+00   
25%    249.50000  1.450000e+07  4.288145e+09     194.500000      3.230000e+02   
50%    498.00000  1.770000e+07  7.760820e+09     729.000000      9.155000e+02   
75%    746.50000  2.460000e+07  1.355470e+10    2667.500000      3.584500e+03   
max    995.00000  2.450000e+08  2.280000e+11  301308.000000      4.057944e+06   

       country_rank  channel_type_rank  video_views_for_the_last_30_days  \
count    879.000000         962.000000                      9.390000e+02   
mean     386.053470         745.719335                      1.756103e+08   
std     1232.244746        1944.386561                      4.163782e+08   
min        1.000000           1.000000                      1.000000e+00   
25%       11.000000          27.000000                      2.013750e+07   
50%       51.000000          65.500000                      6.408500e+07   
75%      123.000000         139.750000                      1.688265e+08   
max     7741.000000        7741.000000                      6.589000e+09   

       lowest_monthly_earnings  highest_monthly_earnings  ...  \
count               995.000000              9.950000e+02  ...   
mean              36886.148281              5.898078e+05  ...   
std               71858.724092              1.148622e+06  ...   
min                   0.000000              0.000000e+00  ...   
25%                2700.000000              4.350000e+04  ...   
50%               13300.000000              2.127000e+05  ...   
75%               37900.000000              6.068000e+05  ...   
max              850900.000000              1.360000e+07  ...   

       highest_yearly_earnings  subscribers_for_last_30_days  created_year  \
count             9.950000e+02                  6.580000e+02    990.000000   
mean              7.081814e+06                  3.490791e+05   2012.630303   
std               1.379704e+07                  6.143554e+05      4.512503   
min               0.000000e+00                  1.000000e+00   1970.000000   
25%               5.217500e+05                  1.000000e+05   2009.000000   
50%               2.600000e+06                  2.000000e+05   2013.000000   
75%               7.300000e+06                  4.000000e+05   2016.000000   
max               1.634000e+08                  8.000000e+06   2022.000000   

       created_date  Gross tertiary education enrollment (%)    Population  \
count    990.000000                               872.000000  8.720000e+02   
mean      15.746465                                63.627752  4.303873e+08   
std        8.777520                                26.106893  4.727947e+08   
min        1.000000                                 7.600000  2.025060e+05   
25%        8.000000                                36.300000  8.335541e+07   
50%       16.000000                                68.000000  3.282395e+08   
75%       23.000000                                88.200000  3.282395e+08   
max       31.000000                               113.100000  1.397715e+09   

       Unemployment rate  Urban_population    Latitude   Longitude  
count         872.000000      8.720000e+02  872.000000  872.000000  
mean            9.279278      2.242150e+08   26.632783  -14.128146  
std             4.888354      1.546874e+08   20.560533   84.760809  
min             0.750000      3.558800e+04  -38.416097 -172.104629  
25%             5.270000      5.590832e+07   20.593684  -95.712891  
50%             9.365000      2.706630e+08   37.090240  -51.925280  
75%            14.700000      2.706630e+08   37.090240   78.962880  
max            14.720000      8.429340e+08   61.92411

**Ответьте на следующие вопросы:**


1.   Какие средние значения в каждой числовой колонке вашего датасета?
2.   Минимальные и максимальные значения в каждой числовой колонке вашего датасета





1.   **Ответ 1.** Средние значения - 498.00000	2.298241e+07	1.103954e+10	9187.125628	5.542489e+05	386.053470	745.719335	1.756103e+08	36886.148281	5.898078e+05	...	7.081814e+06	3.490791e+05	2012.630303	15.746465	63.627752	4.303873e+08	9.279278	2.242150e+08	26.632783	-14.128146 по порядку как в таблице describe
2.   **Ответ 2.** MIN = 1.00000	1.230000e+07	0.000000e+00	0.000000	1.000000e+00	1.000000	1.000000	1.000000e+00	0.000000	0.000000e+00	...	0.000000e+00	1.000000e+00	1970.000000	1.000000	7.600000	2.025060e+05	0.750000	3.558800e+04	-38.416097	-172.104629
MAX =995.00000	2.450000e+08	2.280000e+11	301308.000000	4.057944e+06	7741.000000	7741.000000	6.589000e+09	850900.000000	1.360000e+07	...	1.634000e+08	8.000000e+06	2022.000000	31.000000	113.100000	1.397715e+09	14.720000	8.429340e+08	61.924110	138.252924



##Типизация
Присвойте соответствующим колонкам один из типов (category, float, int)

In [ ]:
df['category']=df['category'].astype('category')
df['subscribers']=df['subscribers'].astype(int)
df['Unemployment rate']=df['Unemployment rate'].astype(float)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 995 entries, 0 to 994
Data columns (total 28 columns):
 #   Column                                   Non-Null Count  Dtype   
---  ------                                   --------------  -----   
 0   rank                                     995 non-null    int64   
 1   Youtuber                                 995 non-null    object  
 2   subscribers                              995 non-null    int64   
 3   video views                              995 non-null    float64 
 4   category                                 949 non-null    category
 5   Title                                    995 non-null    object  
 6   uploads                                  995 non-null    int64   
 7   Country                                  873 non-null    object  
 8   Abbreviation                             873 non-null    object  
 9   channel_type                             965 non-null    object  
 10  video_views_rank                      

Какие признаки вы можете выделить в своем датасете (категориальные, числовые и др., **минимум 3 разных**, помните что можно также выделить признаки более точно, например числовые - порядковые, количественные и т.д.)?
Категориальные - Категории,страна
числовые - просмотры, подписчики
порядковые -ранк, video_view_rank

##Фильтрация
Офильтруйте 3 колонки вашего датасета и выведите результат

In [ ]:
df.loc[df['rank']<500]

rank                          Youtuber  subscribers   video views  \
0       1                          T-Series    245000000  2.280000e+11   
1       2                    YouTube Movies    170000000  0.000000e+00   
2       3                           MrBeast    166000000  2.836884e+10   
3       4        Cocomelon - Nursery Rhymes    162000000  1.640000e+11   
4       5                         SET India    159000000  1.480000e+11   
..    ...                               ...          ...           ...   
494   495  MSA previously My Story Animated     17800000  8.588705e+09   
495   496                  ýýýýýýýýý KIMPRO     17700000  1.920670e+10   
496   497                  Jane ASMR ��     17700000  7.387622e+09   
497   498                  ArianaGrandeVevo     17700000  2.011554e+10   
498   499                      jaanvi patel     17700000  7.912733e+09   

             category                             Title  uploads  \
0               Music                          T-Series    20082   
1    Film & Animation                     youtubemovies        1   
2       Entertainment                           MrBeast      741   
3           Education        Cocomelon - Nursery Rhymes      966   
4               Shows                         SET India   116536   
..                ...                               ...      ...   
494  Film & Animation  MSA Previously My Story Animated       26   
495               NaN                  ýýýýýýýýý KIMPRO     1646   
496               NaN                  Jane ASMR ��     1888   
497             Music                  ArianaGrandeVevo      178   
498    People & Blogs                      jaanvi patel      512   

           Country Abbreviation   channel_type  ...  \
0            India           IN          Music  ...   
1    United States           US          Games  ...   
2    United States           US  Entertainment  ...   
3    United States           US      Education  ...   
4            India           IN  Entertainment  ...   
..             ...          ...            ...  ...   
494  United States           US         People  ...   
495    South Korea           KR         People  ...   
496    South Korea           KR         People  ...   
497  United States           US          Music  ...   
498          India           IN         People  ...   

     subscribers_for_last_30_days  created_year  created_month  created_date  \
0                       2000000.0        2006.0            Mar          13.0   
1                             NaN        2006.0            Mar           5.0   
2                       8000000.0        2012.0            Feb          20.0   
3                       1000000.0        2006.0            Sep           1.0   
4                       1000000.0        2006.0            Sep          20.0   
..                            ...           ...            ...           ...   
494                        2960.0        2021.0            Nov           9.0   
495                     3200000.0        2017.0            Nov          11.0   
496                           NaN        2012.0            Nov          17.0   
497                           NaN        2010.0            Oct          21.0   
498                      900000.0        2018.0            Jul          18.0   

     Gross tertiary education enrollment (%)    Population  Unemployment rate  \
0                                       28.1  1.366418e+09               5.36   
1                                       88.2  3.282395e+08              14.70   
2                                       88.2  3.282395e+08              14.70   
3                                       88.2  3.282395e+08              14.70   
4                                       28.1  1.366418e+09               5.36   
..                                       ...           ...                ...   
494                                     88.2  3.282395e+08              14.70   
495                          

In [ ]:
df.loc[df['category']=='Shows']

rank                                 Youtuber  subscribers   video views  \
4       5                                SET India    159000000  1.480000e+11   
15     16                                 Sony SAB     83000000  1.010000e+11   
26     27                                Colors TV     64600000  6.151091e+10   
67     68                       Masha and The Bear     42400000  2.451902e+10   
89     90                                 Sony PAL     38200000  2.851934e+10   
137   138                                  Mr Bean     32100000  1.060224e+10   
293   294  Kids TV - Nursery Rhymes And Baby Songs     23100000  1.315187e+10   
415   416                                Sony AATH     19600000  1.393002e+10   
440   441                           Masha e o Urso     18900000  8.301731e+09   
491   492                            MGC Playhouse     17900000  1.617453e+10   
632   633                     ýýýýýýýý ýý ýýýýýýýý     15500000  8.265130e+09   
819   820       Kids TV India Hindi Nursery Rhymes     13800000  6.646953e+09   
920   921             ýýýýýýýý ýýýý ýýýýýýýýýýýýýý     12900000  5.585085e+09   

    category                                    Title  uploads  \
4      Shows                                SET India   116536   
15     Shows                                 Sony SAB    71270   
26     Shows                                Colors TV   112915   
67     Shows                       Masha and The Bear     1218   
89     Shows                                 Sony PAL    24089   
137    Shows                                  Mr Bean     3091   
293    Shows  Kids TV - Nursery Rhymes And Baby Songs     3781   
415    Shows                                Sony AATH    18661   
440    Shows                           Masha e o Urso        0   
491    Shows                            MGC Playhouse     1426   
632    Shows                     ýýýýýýýý ýý ýýýýýýýý     1021   
819    Shows       Kids TV India Hindi Nursery Rhymes     1505   
920    Shows             ýýýýýýýý ýýýý ýýýýýýýýýýýýýý     1255   

            Country Abbreviation   channel_type  ...  \
4             India           IN  Entertainment  ...   
15            India           IN  Entertainment  ...   
26            India           IN  Entertainment  ...   
67    United States           US           Film  ...   
89            India           IN  Entertainment  ...   
137  United Kingdom           GB         Comedy  ...   
293   United States           US          Music  ...   
415   United States           US  Entertainment  ...   
440             NaN          NaN            NaN  ...   
491   United States           US      Education  ...   
632    Saudi Arabia           SA           Film  ...   
819           India           IN      Education  ...   
920         Ukraine           UA  Entertainment  ...   

     subscribers_for_last_30_days  created_year  created_month  created_date  \
4                       1000000.0        2006.0            Sep          20.0   
15                      1100000.0        2007.0            Aug           4.0   
26                      1100000.0        2008.0            Jun          13.0   
67                      1000000.0        2014.0            Sep          17.0   
89                       500000.0        2014.0            May           9.0   
137                      100000.0        2006.0            Jan           4.0   
293                      700000.0        2013.0            Aug          30.0   
415                      300000.0        2013.0            Nov          13.0   
440                           2.0        2014.0            Feb          20.0   
491                      100000.0        2009.0            Feb           3.0   
632                      300000.0        2017.0            May          16.0   
819                      100000.0        2014.0            Nov          17.0   
920                      200000.0        2014.0            Jul          17.0   

     Gross tertiary education enrollmen

In [ ]:
df.loc[df['Country']=='India']

rank                  Youtuber  subscribers   video views  \
0       1                  T-Series    245000000  2.280000e+11   
4       5                 SET India    159000000  1.480000e+11   
10     11         Zee Music Company     96700000  5.785629e+10   
15     16                  Sony SAB     83000000  1.010000e+11   
21     22                    Zee TV     70500000  7.313905e+10   
..    ...                       ...          ...           ...   
980   981                  Zee Kids     12400000  7.683670e+09   
981   982          Pari's Lifestyle     12400000  7.741765e+09   
984   985      bharatzkitchen HINDI     12400000  2.315227e+09   
991   992  Free Fire India Official     12300000  1.674410e+09   
994   995              Make Joke Of     12300000  2.129774e+09   

             category                     Title  uploads Country Abbreviation  \
0               Music                  T-Series    20082   India           IN   
4               Shows                 SET India   116536   India           IN   
10              Music         Zee Music Company     8548   India           IN   
15              Shows                  Sony SAB    71270   India           IN   
21      Entertainment                    Zee TV   129204   India           IN   
..                ...                       ...      ...     ...          ...   
980  Film & Animation                  Zee Kids     1212   India           IN   
981    People & Blogs          Pari's Lifestyle      459   India           IN   
984     Howto & Style      bharatzkitchen HINDI      729   India           IN   
991    People & Blogs  Free Fire India Official     1500   India           IN   
994            Comedy              Make Joke Of       62   India           IN   

      channel_type  ...  subscribers_for_last_30_days  created_year  \
0            Music  ...                     2000000.0        2006.0   
4    Entertainment  ...                     1000000.0        2006.0   
10           Music  ...                     1100000.0        2014.0   
15   Entertainment  ...                     1100000.0        2007.0   
21   Entertainment  ...                      900000.0        2005.0   
..             ...  ...                           ...           ...   
980  Entertainment  ...                      100000.0        2012.0   
981         People  ...                           NaN        2018.0   
984          Howto  ...                      100000.0        2016.0   
991          Games  ...                      300000.0        2018.0   
994         Comedy  ...                      100000.0        2017.0   

     created_month  created_date  Gross tertiary education enrollment (%)  \
0              Mar          13.0                                     28.1   
4              Sep          20.0                                     28.1   
10             Mar          12.0                                     28.1   
15             Aug           4.0                                     28.1   
21             Dec          11.0                                     28.1   
..             ...           ...                                      ...   
980            Jul           6.0                                     28.1   
981            Nov          26.0                                     28.1   
984            Aug          10.0                                     28.1   
991            Sep          14.0                                     28.1   
994            Aug           1.0                                     28.1   

       Population  Unemployment rate  Urban_population   Latitude  Longitude  
0    1.366418e+09               5.36       471031528.0  20.593684   78.96288  
4    1.366418e+09               5.36       471031528.0  20.593684   78.96288  
10   1.366418e+09               5.36       471031528.0  20.593684   78.96288  
15   1.366418e+09               5.36       471031528.0  20.593684   78.96288  
21   1.366418e+09               5.36       471031528.0  20.593684 

##Группировка
Сгруппируйте датасет по какому-либо категориальному признаку и выведите среднее значение численных колонок (см. метод mean в pandas)

In [ ]:
df.groupby('Country').mean()

<ipython-input-25-f9dfe299928b>:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.mean is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby('Country').mean()


rank   subscribers   video views       uploads  \
Country                                                                      
Afghanistan           378.000000  2.040000e+07  1.339700e+10      0.000000   
Andorra               664.000000  1.510000e+07  2.400038e+09    159.000000   
Argentina             347.000000  2.526923e+07  1.495503e+10   1326.153846   
Australia             482.777778  1.911111e+07  7.682424e+09    797.666667   
Bangladesh            809.000000  1.390000e+07  1.212958e+10      6.000000   
Barbados               70.000000  4.190000e+07  2.247775e+10     84.000000   
Brazil                583.661290  1.970645e+07  7.761435e+09   2712.548387   
Canada                490.733333  2.426000e+07  1.275470e+10   3116.333333   
Chile                 298.333333  2.906667e+07  9.782519e+09   2257.000000   
China                 508.000000  1.760000e+07  2.977742e+09    128.000000   
Colombia              421.363636  2.404545e+07  1.401798e+10    979.909091   
Cuba                   51.000000  4.630000e+07  2.293663e+10      1.000000   
Ecuador               811.500000  1.385000e+07  2.746231e+09    967.000000   
Egypt                 654.000000  1.530000e+07  4.912370e+09   2968.000000   
El Salvador            53.000000  4.610000e+07  1.032339e+10    543.000000   
Finland               877.000000  1.320000e+07  2.036408e+09    172.000000   
France                610.000000  1.678000e+07  6.111528e+09    445.400000   
Germany               446.500000  1.940000e+07  9.038708e+09   1123.833333   
India                 495.404762  2.565833e+07  1.357767e+10  22191.738095   
Indonesia             528.107143  1.907500e+07  5.393238e+09  24225.750000   
Iraq                  721.000000  1.510000e+07  8.503826e+09   1930.000000   
Italy                 557.000000  1.970000e+07  9.947449e+09    902.500000   
Japan                 387.800000  3.782000e+07  1.713640e+10   3771.200000   
Jordan                423.666667  2.233333e+07  1.769669e+10   2222.333333   
Kuwait                159.000000  3.050000e+07  4.521574e+09    641.000000   
Latvia                389.000000  2.020000e+07  2.091940e+10   1935.000000   
Malaysia              504.000000  1.770000e+07  9.059696e+09   1716.000000   
Mexico                561.424242  1.899697e+07  6.006665e+09   1012.424242   
Morocco               750.000000  1.450000e+07  4.315486e+09    176.000000   
Netherlands           586.333333  1.936667e+07  7.955429e+09   4776.333333   
Pakistan              501.000000  2.590000e+07  1.888224e+10  84657.833333   
Peru                  755.000000  1.440000e+07  1.629801e+09    399.000000   
Philippines           554.916667  2.000833e+07  1.076067e+10  69878.583333   
Russia                513.625000  2.545000e+07  1.456227e+10   1222.375000   
Samoa                 900.000000  1.310000e+07  6.637821e+09     12.000000   
Saudi Arabia          519.666667  1.990000e+07  8.851887e+09   1456.333333   
Singapore             466.333333  1.990000e+07  9.659635e+09  10679.666667   
South Korea           484.000000  2.833529e+07  1.327010e+10   7630.823529   
Spain                 646.454545  1.750455e+07  6.023670e+09   3189.090909   
Sweden                689.750000  1.575000e+07  5.009226e+09    323.250000   
Switzerland           424.000000  1.940000e+07  5.529132e+09  10728.000000   
Thailand              479.388889  2.147222e+07  1.468433e+10  47719.611111   
Turkey                367.000000  2.290000e+07  2.309981e+10   7047.250000   
Ukraine               680.750000  1.570000e+07  7.308166e+09  27516.000000   
United Arab Emirates  485.285714  2.117143e+07  1.076009e+10  26917.428571   
United Kingdom        514.209302  2.133488e+07  1.001318e+10   2018.976744   
United States         482.444089  2.341693e+07  1.179007e+10   4152.105431   
Venezuela             152.000000  3.120000e+07  9.673649e+09     65.000000   
Vietnam               634.666667  1.593333e+07  8.362771e+09   1802.666667   

                      video_views_rank  country_r

##Задание для гениев *
Агрегируйте числовой признак по категориальному признаку с использованием метода **без использования** библиотеки pandas, и вычислите медиану и среднее значение для каждой категории.

In [ ]:
#Агрегация